libc/test/IntegrationTest/test.h

```diff
@@ -58,6 +65,15 @@
 #define ASSERT_NE(val1, val2)                                                  \
   __CHECK_NE(__FILE__, __LINE__, (val1), (val2), true)
 
+////////////////////////////////////////////////////////////////////////////////
+// Errno checks.
+
+#define EXPECT_ERRNO_EQ(VAL) EXPECT_EQ(VAL, static_cast<int>(libc_errno))
+#define ASSERT_ERRNO_EQ(VAL) ASSERT_EQ(VAL, static_cast<int>(libc_errno))
+
+#define EXPECT_ERRNO() EXPECT_NE(static_cast<int>(libc_errno), 0)
```

Comment:
Shall we consistently choose one between `ASSERT_ERRNO_EQ(0)` and `ASSERT_ERRNO()`?  I see that both are used below.  Maybe just pick the former one, and drop `ASSERT_ERRNO()` form, since the former one is essential for other value.

Reply:
To be honest I was not super happy with the naming.
`ASSERT_ERRNO` asserts that the error is **not** 0 (i.e., there is an error).

Maybe we should have `ASSERT_ERRNO_EQ` and `ASSERT_ERRNO_NE`, it would be clearer I think.

Also maybe we don't need the EXPECT and ASSERT versions WDYT?

How about reducing all of these to `ASSERT_ERRNO_EQ` and `ASSERT_ERRNO_NE`?

Reply:
SGTM.

Reply:
Or maybe `ASSERT_ERRNO_EQ` and `ASSERT_ERRNO_FAILURE`?

Reply:
I pushed a a new version, let me know what you think.

Reply:
I've added another variant with `ASSERT_ERRNO_EQ`, `ASSERT_ERRNO_FAILURE` and `ASSERT_ERRNO_SUCCESS`.